File: ruby/TennisGame.rb
class TennisGame

    def initialize(player1Name, player2Name)
        @player1Name = player1Name
        @player2Name = player2Name
        @p1points = 0
        @p2points = 0
        @resultNames = {
                0 => "Love-All",
                1 => "Fifteen-All",
                2 => "Thirty-All",
                3 => "Forty-All",
            }
        @resultNamesSimple = {
                0 => "Love",
                1 => "Fifteen",
                2 => "Thirty",
                3 => "Forty",
        }
    end
        
    def won_point(playerName)
        if (playerName == @player1Name)
            @p1points += 1
        else
            @p2points += 1
        end
    end
    
    def score()
        result = ""
        tempScore=0
        if (@p1points==@p2points)
            result = @resultNames.fetch(@p1points, "Deuce")
        elsif (@p1points>=4 or @p2points>=4)
            minusResult = @p1points-@p2points
            if (minusResult==1)
                result ="Advantage " + @player1Name
            elsif (minusResult ==-1)
                result ="Advantage " + @player2Name
            elsif (minusResult>=2)
                result = "Win for " + @player1Name
            else
                result ="Win for " + @player2Name
            end
        else
            for i in 1..2
                if (i==1)
                    tempScore = @p1points
                else
                    result+="-"
                    tempScore = @p2points
                end
                result += @resultNamesSimple.fetch(tempScore)
            end
        end
        result
    end

end

File: php/TennisGame.php
<?php

class TennisGame
{

    private $player1Name;
    private $player2Name;
    private $p1points;
    private $p2points;
    private $resultNames;
    private $resultNamesSimple;
    
    public function __construct($player1Name, $player2Name)
    {
        $this->player1Name = $player1Name;
        $this->player2Name = $player2Name;
        $this->p1points = 0;
        $this->p2points = 0;
        $this->resultNames = array(
            'Love-All',
            'Fifteen-All',
            'Thirty-All',
            'Forty-All'
        );
        $this->resultNamesSimple = array(
            'Love',
            'Fifteen',
            'Thirty',
            'Forty'
        );
    }

    public function won_point($playerName)
    {
        if ($playerName == $this->player1Name)
            $this->p1points += 1;
        else $this->p2points +=1;
    }

    public function score()
    {
        $result = '';
        $tempScore = 0;
        if ($this->p1points == $this->p2points)
        {
            if ($this->p1points > 3) $result = 'Deuce';
            else $result = $this->resultNames[$this->p1points];
        }
        else if ($this->p1points >= 4 || $this->p2points >= 4)
        {
            $minusResult = $this->p1points - $this->p2points;
            if ($minusResult == 1)
                $result = 'Advantage ' . $this->player1Name;
            else if ($minusResult == -1)
                $result = 'Advantage ' . $this->player2Name;
            else if ($minusResult >= 2)
                $result = 'Win for ' . $this->player1Name;
            else
                $result = 'Win for ' . $this->player2Name;
        }
        else
        {
            for ($i=1; $i <= 2; $i++)
            { 
                if ($i == 1)
                    $tempScore = $this->p1points;
                else
                {
                    $result .= '-';
                    $tempScore = $this->p2points;
                }
                $result .= $this->resultNamesSimple[$tempScore];
            }
        } 
        return $result; 
    }
}